### PR TITLE
Implement ProblemDetails exception handling for AB#14294

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#14294](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14294)

Also implements [AB#14272](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14272) and [AB#14292](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14292)

## Description

- Created Get Patient V2 POC using ProblemDetails for exception handling
- Patient Controller updated to have a 1.0 and 2.0 Get Patient Endpoint call
- Created V2 Patient Service and Client Registries Delegate implementations in Patient project
- Updated HttpWeb.UseHttp to give option to use the development exception page
- In Patient Service V2 replace RequestResult with ApiResult
- In Patient Service V2 use throw ApiPatientException instead of RequestResultError with status Error
- In Client Registries Delegate V2 replace Request Result with ApiResult
- In Client Registries Delegate V2 use throw ApiPatientException instead of populating RequestResultError.Message when status is Error
- In Client Registries Delegate V2 use ApiResult.Warning instead of populating RequestResultError.Message when status is Action Required
- Updated unit test to account for new endpoint in Patient Controller
- Patient Controller returns 200 when call is SUCCESS and when call returns warning messages.  
- Patient Controller returns 404 when patient cannot be returned
- Uncaught exceptions are now handled by Problem Detail implementation

**Get Patient 200:**

Screenshot 2022-11-07 at 2 39 54 PM

**Get Patient 404:**

Screenshot 2022-11-07 at 6 46 22 PM

**Get Patient 500:**

Screenshot 2022-11-07 at 2 41 53 PM

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

No

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Patient/src/Delegates/IClientRegistriesDelegate.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public interface IClientRegistriesDelegate
+    {
+        /// <summary>
+        /// Gets the patient record.
+        /// </summary>
+        /// <param name="type">The oid type value.</param>
+        /// <param name="value">The associated oid type's value to retrieve the patient demographics information.</param>
+        /// <param name="disableIdValidation">Disables the validation on HDID/PHN when true.</param>
+        /// <returns>The patient model wrapped in an api result object.</returns>
+        Task<ApiResult<PatientModel>> GetDemographicsAsync(OidType type, string value, bool disableIdValidation = false);
```
Comment (MikeLyttle) [inline Apps/Patient/src/Delegates/IClientRegistriesDelegate.cs:35]: Suggest renaming parameter `value` to `identifier`.

File: Apps/Patient/src/Controllers/PatientController.cs [hunk 2]
```diff
@@ -59,11 +65,34 @@ public PatientController(IPatientService patientService)
...
+        /// </response>
+        [HttpGet]
+        [Produces("application/json")]
+        [ApiVersion("2.0")]
+        [Route("{hdid}")]
+        [Authorize(Policy = PatientPolicy.Read)]
+        public async Task<OkObjectResult> GetPatientV1(string hdid)
+        {
+            return this.Ok(await this.serviceV2.GetPatient(hdid).ConfigureAwait(true));
+        }
```
Comment (MikeLyttle) [inline Apps/Patient/src/Controllers/PatientController.cs:96]: Based on [this documentation](https://learn.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-6.0#iactionresult-type-1), maybe we should return `Task<IActionResult>` and decorate the method with `[ProducesResponseType]` annotations.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/Patient/src/Delegates/IClientRegistriesDelegate.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/Patient/src/Delegates/IClientRegistriesDelegate.cs:35]: ok

File: Apps/Patient/src/Controllers/PatientController.cs (hunk 2 above)
Comment (BrianMaki) [reply Apps/Patient/src/Controllers/PatientController.cs]: ok

**Pull request merged**